IceCreamShop.playground/Contents.swift

```diff
@@ -1 +1,156 @@
+        var index: Int = 0
+        for f in topFlavors {
+            message += f + ", "
+            if index == topFlavors.count - 2 {
+                message += "and "
+            }
+            
+            index += 1
+        }
+        
+        //account for edge cases, such as if there are no flavors with a rating above 4.0
+        if totalTopFlavors > 0 {
+            print("\(message)")
+        } else {
+            print("There are no top flavors!")
+        }
+        
+    }
+    
+    /*
+     Customers will need a way to order a cone.
+
+     Create a function called orderCone.
+     This should have arguments needed to initialize a Cone, such as a flavor, topping (if desired), and a size.
+     This function should return an optional Cone object. We're going to have it be optional for later use.
+     In this function, create a constant that initializes a new Cone with the arguments passed into the function.
+     Add the price of the cone to the totalSales variable you made in the previous step.
+     Create a string that tells the price of the cone, along with its flavor and topping. NOTE: account for the potential lack of a topping on the Cone in that string by using optional binding (if-let). For example, the string could say "Your mint ice cream with chocolate chips is 3.99", or "Your vanilla ice cream is 5.99." Print the string.
+     Finally, return the cone you initialized.
+     */
+    func orderCone(flavor: Flavor, topping: String?, size: Size) -> Cone? {
+        //In the orderCone function, check to make sure the flavor that the person requested exists on the menu.
+        guard flavorIsOnMenu(flavor: flavor) else { return nil }
+        
+        let myCone = Cone(flavor: flavor, topping: topping, size: size)
+        self.totalSales += myCone.size.rawValue
+        
+        if let myTopping = myCone.topping {
+            print("Your \(myCone.flavor) ice cream with \(myTopping) is \(myCone.size.rawValue)")
+        } else {
+            print("Your \(myCone.flavor) ice cream is \(myCone.size.rawValue)")
+        }
+        return myCone
+    }
+    
+    func flavorIsOnMenu(flavor: Flavor) -> Bool {
+        for f in flavors {
+            if f.name == flavor.name {
+                return true
+            }
+        }
+        return false
+    }
+}
+
+//At the bottom of the playground, create a few Flavor constants, an array of sizes, and an array of toppings.
+let flavorMint: Flavor = Flavor(name: "Mint", rating: 5)
+let flavorChoc: Flavor = Flavor(name: "Chocolate", rating: 8)
+let flavorVan: Flavor = Flavor(name: "Vanilla", rating: 6)
+let sizes: [Size] = [.small, .medium, .large]
+let toppings: [String] = ["Candy","Sprinkles","Peanuts"]
+
+
+//Use the constants you just made to initialize a new IceCreamShop constant.
+let myShop: IceCreamShop = IceCreamShop(flavors: [flavorVan,flavorChoc,flavorMint], toppings: toppings, sizes: sizes)
```

Comment:
Arrays should have a single space after each comma when the elements are on one line.